File: benchmark/csharp/task6_fileio.cs
using System.IO;
using System.Linq;

void CountWords(string inputFile, string outputFile)
{
    var counts = File.ReadAllText(inputFile)
        .Split()
        .Where(w => !string.IsNullOrEmpty(w))
        .GroupBy(w => w)
        .ToDictionary(g => g.Key, g => g.Count());

    File.WriteAllLines(outputFile, counts.Select(kv => $"{kv.Key}: {kv.Value}"));
}

File: benchmark/visualbasic/task6_fileio.vb
Imports System.IO
Imports System.Linq

Module WordCount
    Sub CountWords(inputFile As String, outputFile As String)
        Dim counts = File.ReadAllText(inputFile) _
            .Split() _
            .Where(Function(w) Not String.IsNullOrEmpty(w)) _
            .GroupBy(Function(w) w) _
            .ToDictionary(Function(g) g.Key, Function(g) g.Count())

        File.WriteAllLines(outputFile, counts.Select(Function(kv) $"{kv.Key}: {kv.Value}"))
    End Sub
End Module
